docs/Agent/user-guide/man_page.md

```diff
@@ -1 +1,472 @@
+
+`-g`, `--group`, `--groups` STR  
+List all the triggers registered in the <`STR`> group.
+
+`--help`  
+Show this message and exit.
+
+---
+
+### pbench-move-result
+
+---
+
+**NAME**  
+`pbench-move-results`, Move all results to 0.69 Pbench Server.
+
+**SYNOPSIS**  
+`pbench-move-results [--help] [--user=<user>] [--controller=<controller>] [--prefix=<path>] [--xz-single-threaded] [--show-server]`
+
+**DESCRIPTION**  
+This command can only be used for a 0.69 Pbench Server destination, and requires a `/opt/pbench-agent/id_rsa` file with the private SSH key of the server's pbench account.
+
+**OPTIONS**  
+`--user`  
+This option value is required if not provided by the
+`PBENCH_USER` environment variable; otherwise, a value provided
+on the command line will override any value provided by the
+environment.
+
+`--controller`
+This option may be used to override the value
+provided by the `PBENCH_CONTROLLER` environment variable; if
+neither value is available, the result of `hostname -f` is used.
+(If no value is available, the command will exit with an error.)
+
+`--prefix`  
+This option allows the user to specify an optional
+directory-path hierarchy to be used when displaying the result
+tar balls on the pbench server.
+
+`--show-server`  
+This will not move any results, but resolve and
+then display the pbench server destination for results.
+
+`--xz-single-threaded`  
+This will force the use of a single
+thread for locally compressing the result tar balls.
+
+`--help`  
+Show this message and exit.
+
+---
+
+### pbench-register-tool
+
+---
+
+**NAME**  
+`pbench-register-tool`, Registers the specified tools
+
+**SYNOPSIS**  
+`pbench-register-tool --name=<tool-name> [--group=<group-name>] [--no-install] [--persistent | --transient] [--remotes=<remote-host>[,<remote-host>]] [--labels=<label>[,<label>]] -- <tool-specific-options> [--help]`
+
+`pbench-register-tool --name=<tool-name> [--group=<group-name>] [--no-install] [--persistent | --transient] [--remotes=@<remotes-file>] -- <tool-specific-options>`
+
+**DESCRIPTION**  
+Register the tools that are specified .
+List of available tools:
+
+**Transient**
+- blktrace
+- 	bpftrace
+- 	cpuacct
+- 	disk
+- 	dm-cache
+- 	docker
+- 	docker-info
+- 	external-data-source
+- 	haproxy-ocp
+- 	iostat
+- 	jmap
+- 	jstack
+- 	kvm-spinlock
+- 	kvmstat
+- 	kvmtrace
+- 	lockstat
+- 	mpstat
+- 	numastat
+- 	oc
+- 	openvswitch
+- 	pcp-transient
+- 	perf
+- 	pidstat
+- 	pprof
+- 	proc-interrupts
+- 	proc-sched_debug
+- 	proc-vmstat
+- 	prometheus-metrics
+- 	qemu-migrate
+- 	rabbit
+- 	sar
+- 	strace
+- 	sysfs
+- 	systemtap
+- 	tcpdump
+- 	turbostat
+- 	user-tool
+- 	virsh-migrate
+- 	vmstat
+
+ **Persistent**
+- node-exporter
+- dcgm
+- pcp
+
+**OPTIONS**  
+`--name` STR
+This can be used register a specific tool by name.
+For a list of tool-specific options, run:
+/opt/pbench-agent/tool-scripts/<`tool-name`> --help
+
+`[--persistent | --transient]`  
+This can be used to specify tool run type.
```

Comment:
I remember that this was intended to support tools available both as persistent and transient. I think the only example of that now is `pcp`, which has both `pcp` and `pcp-transient`. That only really makes sense if the tool selection allows prefix matching so that "pcp" alone is ambiguous... but while I don't remember if the lookup works that way, I know that was the original intent for the options.

Reply:
My memories are similar to yours...however:
```
$ ./contrib/containerized-pbench/pbench pbench-register-tool --name pcp --transient
[error][2023-06-05T18:44:17.559481493] Could not find pcp-transient in /opt/pbench-agent/tool-scripts/: has this tool been integrated into the pbench-agent?
usage:
pbench-register-tool --name=<tool-name> [--group=<group-name>] [--no-install] [--persistent] [--transient] [--remotes=<remote-host>[,<remote-host>]] [--labels=<label>[,<label>]] -- [all tool specific options here]
pbench-register-tool --name=<tool-name> [--group=<group-name>] [--no-install] [--persistent] [--transient] [--remotes=@<remotes-file>] -- [all tool specific options here]
  ...
```

Reply:
And...
```
$ ./contrib/containerized-pbench/pbench pbench-register-tool --name pcp --permanent

--name pcp --permanent
pbench-register-tool: you specified an invalid option

usage:
  ...
```

But, with no qualifier, it works:
```
$ ./contrib/containerized-pbench/pbench pbench-register-tool --name pcp
"pcp" tool is now registered for host "pbench-ctr-build" in group "default"
```

Reply:
The other thing that I noted is that `user-tool` is only available as a "transient" tool...but...I'm not sure that that is how it actually gets run (I can't remember...maybe it does get cycled on each iteration...I can't remember).

Reply:
I'm pretty sure that without the qualifier you're using the persistent `pcp`. The `get-internal-tool` script tries to append the `-<tool-type>` suffix to allow looking up the non-default variation; however it appears that something's going wrong where it claims it can't find `tool-scripts/pcp-transient` ... which definitely exists, at least in the source.

Ah: there must be a packaging bug, because `pcp-transient` isn't there:

```bash
(.venv39) dbutenho 15:02 torelay:torelay ./contrib/containerized-pbench/pbench bash
[root@dbutenho /]# ls /opt/pbench-agent/tool-scripts/pcp*
/opt/pbench-agent/tool-scripts/pcp
```

Reply:
The really weird thing about these options is that they come *after* the tool name, since they work by modifying the search path of the `name` variable that's already set. That seems really counter-intuitive and we probably should explain that. It's not a "mode switch", but an immediate *modifier* of the current state. That's ... weird and in my experience very unusual.

And ... yeah, as Webb said, given the current set of tools this is necessary only if you want a "transient" pcp; we have no current use for `--persistent` at all. It's fine I suppose that the command was designed to be general, but maybe this is all just unnecessary confusion. Especially, perhaps, since the current Agent packaging seems to omit the only tool (`pcp-transient`) for which either of these options matter! 😆

Reply:
I have no problem with us documenting `--persistent`, even though it is functionally the default for the `pcp` tool and doesn't have any other obvious use.  (Note, I blew it when [I tested above](https://github.com/distributed-system-analysis/pbench/pull/3442/files#r1218460121), and I've corrected it, now.)

And, I think we should document `--transient` even though it doesn't actually "work" at the moment (see [this comment](https://github.com/distributed-system-analysis/pbench/pull/3442#discussion_r1219987101)).

I'm not sure exactly what we should _say_ about them, though.  😒 

I'm not excited about having to point out that the option order matters.  Now that that is on our list of things to fix, I'm inclined to gloss over that point, on the assumption that the options are properly ordered in the synopsis and that users will figure it out when they get the error if they reverse the order.

So, I guess we could say something like,
> `[--persistent | --transient]`  
> For tools which can be run as either "transient" (where they are started and stopped on each iteration) or as "persistent" (where they are started before the first iteration and run continuously over all iterations), these options determine how the tool will be run.  (Most tools are configured to run as either or the other, so these options are not necessary in those cases, and specifying the wrong one will produce an error.)